### PR TITLE
USHIFT-6362: Add iso deployment for rpmostree release scenarios

### PR DESCRIPTION
**Pull request by kasturinarra**

### CODE REVIEW

**Comment by kasturinarra** [comment]
@agullon could you please help review this as well ? thanks !!

File: test/bin/build_images.sh [hunk 1]
```diff
@@ -389,6 +389,26 @@ do_group() {
...
+                blueprint_toml_file="${IMAGEDIR}/blueprints/$(basename "${toml_basename}")"
+                if [ -f "${blueprint_toml_file}" ] && [ -s "${blueprint_toml_file}" ]; then
+                    sudo composer-cli blueprints push "${blueprint_toml_file}"
+                else
+                    echo "WARNING: Blueprint file ${blueprint_toml_file} is missing or empty, skipping ISO build"
+                    record_junit "${groupdir}" "${image_installer}" "compose" "SKIPPED"
+                    continue
+                fi
+            fi
+
```
Comment (agullon) [inline test/bin/build_images.sh]: If I understand this block correctly, what it does is to push into the blueprints registry the missing rpm-ostree ref before creating iso files from those images.

I think we don't need this, because this is done by the groups dir structure: all the `*.image-installer` files (an iso file will be created for every file like this) must be in a group that is after the `*.toml` files (a rpm-ostree commit will be created for every file like this).

File: test/scenarios/releases/el96-lrel@iso-standard2.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (agullon) [inline test/scenarios/releases/el96-lrel@iso-standard2.sh:1]: I think it's better to replace `test/scenarios/releases/el96-lrel@standard2.sh` with this one, to follow the same approach done in bootc scenarios.

File: test/scenarios/releases/el96-lrel@iso-standard1.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (agullon) [inline test/scenarios/releases/el96-lrel@iso-standard1.sh:1]: I think it's better to replace `test/scenarios/releases/el96-lrel@standard1.sh` with this one, to follow the same approach done in bootc scenarios.

File: test/bin/build_images.sh (hunk 1 above)
Comment (ggiguash) [reply test/bin/build_images.sh]: +1 on this. We should not need to change the build script.

File: test/image-blueprints/layer4-release/group4/rhel96-brew-zstream-with-optionals.image-installer [hunk 4]
```diff
@@ -0,0 +1 @@
+rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION }}-zstream
```
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-zstream-with-optionals.image-installer]: This file must be in the next group (create a new `group4` if needed). This is because all builds in the same group will be built in parallel and this one can not be created until `test/image-blueprints/layer4-release/group3/rhel96-brew-zstream-with-optionals.toml` is successfully created.

File: test/image-blueprints/layer4-release/group4/rhel96-brew-rc-with-optionals.image-installer [hunk 5]
```diff
@@ -0,0 +1 @@
+rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION }}-rc
```
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-rc-with-optionals.image-installer]: This file must be in the next group (create a new `group4` if needed). This is because all builds in the same group will be built in parallel and this one can not be created until `test/image-blueprints/layer4-release/group3/rhel96-brew-rc-with-optionals.toml` is successfully created.

File: test/image-blueprints/layer4-release/group4/rhel96-brew-ec-with-optionals.image-installer.disabled [hunk 6]
```diff
@@ -0,0 +1 @@
+rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION }}-ec
```
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-ec-with-optionals.image-installer.disabled]: This file must be in the next group (create a new `group4` if needed). This is because all builds in the same group will be built in parallel and this one can not be created until `test/image-blueprints/layer4-release/group3/rhel96-brew-ec-with-optionals.toml` is successfully created.

File: test/bin/build_images.sh (hunk 1 above)
Comment (kasturinarra) [reply test/bin/build_images.sh]: with out this i see iso image building fails complaining no `blueprint` exists. error here: ERROR: UnknownBlueprint: Unknown blueprint name: rhel-9.6-microshift-brew-optionals-4.21-ec

https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_microshift/5996/pull-ci-openshift-microshift-main-e2e-aws-tests/2005942455337226240

Comment (agullon) [reply test/bin/build_images.sh]: afaik those errors will be gone if you move the 3 `*.image-installer` files into the next group.

because both image-installer and toml file are in the same group they are built in parallel. And image-installer complains that the blueprint does not exist yet. To wait for toml file to be built before the image-installer we use groups order.

File: test/image-blueprints/layer4-release/group4/rhel96-brew-zstream-with-optionals.image-installer (hunk 4 above)
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-zstream-with-optionals.image-installer]:
```suggestion
{{- if and (env.Getenv "BREW_Y0_RELEASE_VERSION" "") (env.Getenv "BREW_Y1_RELEASE_VERSION" "") -}}
rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION }}-zstream
{{- end -}}
```

File: test/image-blueprints/layer4-release/group4/rhel96-brew-rc-with-optionals.image-installer (hunk 5 above)
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-rc-with-optionals.image-installer]:
```suggestion
{{- if and (env.Getenv "BREW_RC_RELEASE_VERSION" "") (env.Getenv "BREW_Y1_RELEASE_VERSION" "") -}}
rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION}}-rc
{{- end -}}
```

File: test/image-blueprints/layer4-release/group4/rhel96-brew-ec-with-optionals.image-installer.disabled (hunk 6 above)
Comment (agullon) [inline test/image-blueprints/layer4-release/group4/rhel96-brew-ec-with-optionals.image-installer.disabled]:
```suggestion
{{- if and (env.Getenv "BREW_EC_RELEASE_VERSION" "") (env.Getenv "BREW_Y1_RELEASE_VERSION" "") -}}
rhel-9.6-microshift-brew-optionals-4.{{ .Env.MINOR_VERSION }}-ec
{{- end -}}
```

File: test/bin/build_images.sh (hunk 1 above)
Comment (agullon) [reply test/bin/build_images.sh]: I think @kasturinarra is right about this. There's a corner case when no toml files and only image-installer files are in a group The do_group() exists [here](https://github.com/openshift/microshift/blob/main/test/bin/build_images.sh#L303-L311) so [the block](https://github.com/openshift/microshift/blob/main/test/bin/build_images.sh#L372-L401) to build the iso from image-installer files is never executed.
So I suggest to move the [block](https://github.com/openshift/microshift/blob/main/test/bin/build_images.sh#L303-L311) into line 339 to push all the blueprints before running the should_skip()

File: test/bin/build_images.sh [hunk 7]
```diff
@@ -337,6 +327,16 @@ do_group() {
...
 
+        # Check if the image for this blueprint already exists, in
+        # case it was downloaded from the cache.
+        if ostree summary --view --repo="${IMAGEDIR}/repo" | grep -q " ${blueprint}\$"; then
+            echo "Found ${blueprint} in existing images"
+            if should_skip "${blueprint}"; then
+                record_junit "${groupdir}" "${template}" "compose" "SKIPPED"
+                continue
+            fi
+        fi
```
Comment (ggiguash) [inline test/bin/build_images.sh]: Should we put this block before resolving dependencies?

Comment (ggiguash) [reply test/bin/build_images.sh]: /hold

File: test/bin/build_images.sh [hunk 8]
```diff
@@ -300,7 +300,15 @@ do_group() {
...
-        # Check if the image for this blueprint already exists, in
+        if sudo composer-cli blueprints list | grep -q "^${blueprint}$"; then
+            echo "Removing existing definition of ${blueprint}"
+            sudo composer-cli blueprints delete "${blueprint}"
+        fi
+
+        echo "Loading new definition of ${blueprint}"
+        sudo composer-cli blueprints push "${blueprint_file}"
+
+	# Check if the image for this blueprint already exists, in
```
Comment (ggiguash) [inline test/bin/build_images.sh]: Nit: Tabulation seems wrong.

/unhold

Comment (kasturinarra) [reply test/bin/build_images.sh]: thanks, adjusted.

**Comment by kasturinarra** [comment]
/override ci/prow/e2e-aws-tests-bootc ci/prow/e2e-aws-tests-bootc-periodic-arm ci/prow/e2e-aws-tests-bootc-periodic ci/prow/e2e-aws-tests-bootc-release ci/prow/e2e-aws-tests-periodic ci/prow/e2e-aws-tests-bootc-release-arm ci/prow/e2e-aws-tests-release-arm ci/prow/e2e-aws-tests-release
